backend/src/services/tenant/tenant.service.ts

```diff
@@ -32,6 +34,33 @@ class Tenant {
     };
   }
 
+  public async updateTenantById({
+    id,
+    name,
+  }: {
+    id: string;
+    name: string;
+  }): Promise<EAMTenantByIdResponseDto | null> {
+    const tenant = await this.#tenantRepository.getById(id);
+    if (!tenant) {
+      return null;
+    }
+    if (tenant.name === name) {
+      throw new TenantError({
+        status: HttpCode.DENIED,
+        message: ExceptionMessage.TENANT_NAME_EXISTS,
+      });
+    }
```

Comment:
It won't work. Consider an example where we have two tenants:
tenant1 - id1
tenant2 - id2
PUT /id1 with name tenant2
```suggestion
    const tenantWithName = await repo.getByName(name);
    if (tenantWithName !== null) {
      throw new TenantError({
        status: HttpCode.DENIED,
        message: ExceptionMessage.TENANT_NAME_EXISTS,
      });
    }
```